packages/open-truss/src/uqi/example/fake-client-uqi.ts

```diff
@@ -0,0 +1,56 @@
+import Iterator from '../iterator'
+import uqi, {
+  type UqiClient,
+  type UqiContext,
+  type UqiResult,
+  type UqiMappedType,
+} from '../uqi'
+import createFakeClient, {
+  type FakeClient,
+  type FakeClientConfig,
+} from './fake-client'
+
+export default async function (config: FakeClientConfig): Promise<UqiClient> {
+  const typeMappings: Record<string, UqiMappedType> = {
```

Comment:
That's a little annoying you have to declare the type here instead of having it be implicit. I feel like typescript should be smarter about knowing when an untyped object adheres to the interface of the function you are passing it to.

Reply:
Maybe I did it un-necessarily? Not sure, maybe you could hack on it a bit to see if there is a better way?

Reply:
I tried removing the type locally and it wasn't happy :/